File: 2-ad-hoc/1087.py
end=None
def moveQueen(x,y):
    for i in range(2):
        for j in range(8):
            if (x*i+j*int(not i))*10+y*int(not i)+j*i==end:return True
    auxX=x-y
    auxY=0;
    while(auxX<8 and auxY<8):
        if(auxX*10+auxY==end):return True
        auxX+=1
        auxY+=1
    auxX=x-(7-y)
    auxY=7
    while(auxX<8 and auxY>-1):
        if(auxX*10+auxY==end):return True
        auxX+=1
        auxY-=1
    return False
while((x:=list(map(int,input().split())))and x[0] and x[1] and x[2] and x[3]):
    if x[0]==x[2] and x[1]==x[3]:
        print('0')
        continue 
    end=(x[2]-1)*10+x[3]-1
    if moveQueen(x[0]-1,x[1]-1):print('1')
    else: print('2')

File: 2-ad-hoc/1087.c
#include<stdio.h>
//
int end;
int moveQueen(int x,int y){
	for(int i=0;i<2;++i){
		int subX=i==0?1:0,subY=i==0?0:1;
		for(int j=0;j<8;++j)if((x*subY+j*subX)*10+y*subX+j*subY==end)return 1;
	}
	int auxX=x-y,auxY=0;
	for(;auxX<8&&auxY<8;++auxX,++auxY)if(auxX*10+auxY==end)return 1;
	auxX=x-(7-y);auxY=7;
	for(;auxX<8&&auxY>-1;++auxX,--auxY)if(auxX*10+auxY==end)return 1;
	return 0;
}

int main(){
	int x1,x2,y1,y2;
	while(scanf("%i %i %i %i",&x1,&y1,&x2,&y2)&&x1&&x2&&y1&&y2){
		if(x2==x1&&y1==y2){
			printf("0\n");
			continue;
		}
		end=(x2-1)*10+y2-1;
		if(moveQueen(x1-1,y1-1))printf("1\n");
		else printf("2\n");
	}
	return 0;
}
